src/tcp/flow.ml

```diff
@@ -728,10 +717,8 @@ struct
 
   (* Construct the main TCP thread *)
   let connect ip clock =
-    (* XXX: I've no clue why this is the way it is (10000 + Random
-       ~bound:10000) -- hannes *)
     let localport =
-      10000 + (Randomconv.int (fun x -> Random.generate x) ~bound:10000)
+      1024 + (Randomconv.int ~bound:(0xFFFF - 1024) Random.generate)
```

Comment:
This patch looks like a definite improvement to me!

Out of curiosity, do we have to choose local ports > 1024? Is this a Unix convention or a TCP convention? (No big deal, I'm just interested)

Reply:
oh, of course any port number is fine, according to the RFCs[tm].  in practise, there are servers which refuse to service you if your source port range is < 1024 (I experienced this with some authoritative nameservers) - thus I'd (for now at least) avoid to use something < 1024 -- but then our UDP implementation takes any source port -- maybe we should unify these two expressions!?